llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -908,8 +908,13 @@ static void simplifyRecipe(VPRecipeBase &R, VPTypeAnalysis &TypeInfo) {
         unsigned ExtOpcode = match(R.getOperand(0), m_SExt(m_VPValue()))
                                  ? Instruction::SExt
                                  : Instruction::ZExt;
-        auto *VPC =
-            new VPWidenCastRecipe(Instruction::CastOps(ExtOpcode), A, TruncTy);
+        VPSingleDefRecipe *VPC;
```

Comment:
Worth a comment that `UnderlyingExt`, having distinct return type, is recorded only to support legacy cost computation?

Reply:
Should be clearer in the updated version, thanks!